

syntax = "proto3";
package pool;

import "common.proto";

service ModelPool {
    rpc push_model(stream PushRequest) returns (PushResponse);
    rpc pull_model(PullRequest) returns (stream PullResponse);
}

message ModelChunk {
    int32 total_chunks = 2;
    int32 chunk_idx = 3;
    bytes chunk_data = 4 ;
}

message PushRequest {
    common.ModelComponentId model_component_id = 1;
    ModelChunk model_chunk = 2;
}

message PushResponse {
    
}

message PullRequest {
    common.ModelComponentId model_component_id = 1;
}

message PullResponse {
    ModelChunk model_chunk = 1;
}

